agent/ecscni/netconfig_linux.go

```diff
@@ -169,3 +171,46 @@ func NewAppMeshConfig(appMesh *appmesh.AppMesh, cfg *Config) (string, *libcni.Ne
 
 	return defaultAppMeshIfName, networkConfig, nil
 }
+
+// NewServiceConnectNetworkConfig creates a new ServiceConnect CNI network configuration
+func NewServiceConnectNetworkConfig(
+	scConfig *serviceconnect.Config,
+	enableIPv4, enableIPv6 bool,
+	cfg *Config) (string, *libcni.NetworkConfig, error) {
+	var ingressConfig []IngressConfigJSONEntry
+	for _, ic := range scConfig.IngressConfig {
+		newEntry := IngressConfigJSONEntry{
+			ListenerPort: ic.ListenerPort,
+		}
```

Comment:
Why cant we simply copy the Ingress Config entry?

Reply:
Not sure I follow - the `IngressConfigJSONEntry` is a different struct from `scConfig.IngressConfig` entries. Former is the internal model for CNI while latter is the SC API model.

Reply:
I was suggesting that we could directly use the scConfig.IngressConfig given it only has some additional fields and they would be ignored anywhere. But given that the number of entries in here will be less. it might not matter.

Reply:
Yeah I think I'd rather keep it separate. The EgressConfig that we pass to CNI plugin will have to be different than the API model anyway (to include `RedirectMode` for example), so it's probably better to keep IngressConfig separate as well.